### ISSUE TITLE
 template-curly-spacing fix should not remove comments inside the curly braces.

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#12744]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.13.0
* **npm Version:** v6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
 "parserOptions": { "ecmaVersion": 6 },
  rules: {
   "template-curly-spacing": ["error", "never"]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

* .eslintrc.js

```js
module.exports = {
 "parserOptions": { "ecmaVersion": 6 },
  rules: {
   "template-curly-spacing": ["error", "never"]
  }
}
```

* test.js

```js
const foo = `  ${  bar  /*comment*/}   `;
```

Run eslint with `--fix` option.
```bash
eslint --fix test.js
```

**What did you expect to happen?**

It should not remove comments(`/*comment*/`) even it has an error.

**What actually happened? Please include the actual, raw output from ESLint.**

```js
const foo = `  ${bar}   `;
```
It removes comments inside the curly braces.`{`,`}`. Here is a [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgdGVtcGxhdGUtY3VybHktc3BhY2luZzogW1wiZXJyb3JcIiwgXCJuZXZlclwiXSovXG5jb25zdCBmb28gPSBgICAkeyAgYmFyICAvKmNvbW1lbnQqL30gICBgOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Are you willing to submit a pull request to fix this bug?**

Yes, If it is a bug.

**Comment by mdjermanovic** [comment]
It's very possible that some other spacing rules have the same bug.

Spacing rules that don't have this bug treat comments as tokens.

That's one option, another is to defer to a new `space-around-comment` rule (as @mysticatea mentioned in [this comment](https://github.com/eslint/eslint/pull/12216#discussion_r324020108)) in these situations.
